src/regmod/utils.py

```diff
@@ -8,8 +8,32 @@
 from xspline import XSpline
 
 
-def default_vec_factory(vec: Any, size: int, default_value: float,
+def default_vec_factory(vec: Any,
+                        size: int,
+                        default_value: float,
                         vec_name: str = 'vec') -> np.ndarray:
+    """Validate or create the vector.
+
+    Parameters
+    ----------
+    vec : Any
+        Vector to be validated. When it is ``None``, it will use the
```

Comment:
I think you only need single ticks when referring to variables, e.g., `None` rather than ``None``

Reply:
Thanks very much! Just changed to single ticks. Cannot recall where I learn to use double ticks, but I think single is the way to go :)

Reply:
@kels271828 Here is the reference :)
https://stackoverflow.com/questions/22256995/restructuredtext-in-sphinx-and-docstrings-single-vs-double-back-quotes-or-back